src/benchmark/cases/mergesort.js

```diff
@@ -0,0 +1,108 @@
+function copyArray(destination, source, includingStartIndex, excludingEndIndex) {
+    var i = includingStartIndex;
+
+    while (i < excludingEndIndex) {
+        destination[i] = source[i];
+        i++;
+    }
+
+    return destination;
+}
+
+function mergeSort(array, workspaceArray, left, right) {
+    if (right - left == 1) {
+        return workspaceArray;
+    }
+
+    var middle = (left + right) / 2;
+    mergeSort(array, workspaceArray, left, middle);
+    mergeSort(array, workspaceArray, middle, right);
+
+    merge(array, workspaceArray, left, middle, right);
+
+    return workspaceArray;
+}
+
+function merge(array, workspaceArray, left, middle, right) {
+    var indexLeft = left;
+    var indexRight = middle;
+    var index = left;
+
+    while (indexLeft < middle && indexRight < right) {
+        if (array[indexLeft] < array[indexRight]) {
+            workspaceArray[index] = array[indexLeft];
+            indexLeft++;
+        } else {
+            workspaceArray[index] = array[indexRight];
+            indexRight++;
+        }
+
+        index++;
+    }
+
+    while (indexLeft < middle) {
+        workspaceArray[index] = array[indexLeft];
+        indexLeft++;
+        index++;
+    }
+
+    while (indexRight < right) {
+        workspaceArray[index] = array[indexRight];
+        indexRight++;
+        index++;
+    }
+
+    copyArray(array, workspaceArray, left, right);
+    return workspaceArray;
```

Comment:
Is there a reason why you return the `workspaceArray` here?

Reply:
Not really no. I just return something because it is necessary for the transpiler.